gem/backends.py

```diff
@@ -56,18 +56,30 @@ def _update_user_from_claims(user, claims):
     # Synchronise the roles that the user has.
     # The list of roles may contain more or less roles
     # than the previous time the user logged in.
-    roles = set(claims.get("roles", []))
-
-    # If the user has any role, we assume that it is a superuser while
-    # the permissions are being integrated with the Auth Service
-    # (Currently all admin users are superusers)
-    # This is to get logging in working on Core QA site and will change
-    if roles:
-        wagtail_permission = Permission.objects.get(
-            content_type__app_label='wagtailadmin', codename='access_admin')
-        user.user_permissions.add(wagtail_permission)
-        user.is_superuser = True
-        user.save()
+    auth_service_roles = set(claims.get("roles", []))
+    wagtail_groups = set(group.name for group in user.groups.all())
+
+    # If the user has any role, add the wagtail group equivalent
+    # to that role to the user
+    if auth_service_roles:
+        groups_to_add = auth_service_roles - wagtail_groups
+        groups_to_remove = wagtail_groups - auth_service_roles
+        for group_name in groups_to_add:
+            if group_name == 'Product Tech Admin':
```

Comment:
Perhaps "Product Tech Admin" should be defined as a constant somewhere rather than just as a raw value in the middle of a function scope.